test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -1250,6 +1250,35 @@ public Builder testUI(WindowCreator windowCreator) {
             return this;
         }
 
+        /**
+         * Adds an implementation of {@link PositionWindows PositionWindows}
+         * which the framework will use to position multiple test UI windows.
+         *
+         * @param positionWindows an implementation of {@code PositionWindows}
+         *                        to position multiple test UI windows
+         * @return this builder
+         * @throws IllegalArgumentException if the {@code positionWindows}
+         *              parameter is {@code null}
+         * @throws IllegalStateException if the {@code positionWindows} field
+         *              is already set, or if neither a windows list creator
+         *              nor a list of test windows has been provided
+         */
+        public Builder positionTestUI(PositionWindows positionWindows) {
+            if (positionWindows == null) {
+                throw new IllegalArgumentException("positionWindows parameter can't be null");
+            }
+            if (this.positionWindows != null) {
+                throw new IllegalStateException("PositionWindows is already set");
+            }
+            if (windowListCreator == null || testWindows == null) {
+                throw new IllegalStateException("Neither a windows list creator "
+                                                + "nor a list of test windows "
+                                                + "has been provided");
+            }
```

Comment:
@aivanov-jdk Running into exception when this check is added here. I believe this check needs to be done after positionWindows is assigned and instructionUIHandler = new InstructionUIHandler(); is instantiated. 
  
Since this check is already added in [validate()](https://github.com/openjdk/jdk/blob/418bb42b95b177f5f31f756054d0dd83740c6686/test/jdk/java/awt/regtesthelpers/PassFailJFrame.java#L1475C1-L1479C18) , I'm removing it from here.

Reply:
The condition should've been `&&`: the error message is correct, the exception was supposed to be thrown if both `windowListCreator` and `testWindows` is null. The condition with `||` is always `true`.

Yet I agree to remove this condition from here as it enforces the order of calls, which is not strictly required, even though it would common to add a window creator before setting positioning.